docs/src/proposals/rip-curl.md

```diff
@@ -0,0 +1,56 @@
+# RiP Curl: low-latency, transaction-oriented RPC
+
+## Problem
+
+Solana's initial RPC implementation was created for the purpose of allowing
+users to confirm transactions that had just recently been sent to the cluster.
+It was designed with memory usage in mind such that any validator should be
+able to support the API without concern of DoS attacks.
+
+Later down the line, it became desirable to use that same API to support the
+Solana explorer. The original design only supported minutes of history, so we
+changed it to instead store transaction statuses in a local RocksDB instance
+and offer days of history. We then extended that to 6 months via BigTable.
+
+With each modification, the API became more suitable for explorers and less
+appealing for transaction processing. The clients poll for transaction status
+instead of being notified, giving the false impression of higher confirmation
+times. Furthermore, what clients can poll for is limited, preventing them from
+making reasonable real-time decisions, such as recognizing a transaction is
+confirmed as soon as particular, trusted validators vote on it.
+
+## Proposed Solution
+
+A web-friendly, transaction-oriented, streaming API built around the
+validator's ReplayStage.
+
+Improved client experience:
+
+* Support connections directly from JavaScript apps.
+* Clients can be notified of confirmation in optimal time.
+* Clients can be notified in real-time of details such as votes and the voter's
+  stake weight.
+* Clients can be notified when the heaviest fork changes, if it affects the
+  transactions confirmation count.
+
+Easier for validators to support:
+
+* Each validator supports some number of concurrent connections and otherwise
+  has no significant resource constraints.
+* Transaction status is never stored in memory and cannot be polled for.
```

Comment:
Are you suggesting here that we wouldn't have a history query API for transaction status? How would the explorer know transaction status?

Reply:
What happens if a websocket connection needs to reconnect? Or if the connected node goes offline? If a client is hitting a load balancer, they may get a different node.

Reply:
Not suggesting dropping any APIs. I'm only thinking about potentially useful server-side information that isn't available to clients in real-time.

Regarding broken web socket connections, I'm thinking that the user would recover using the historical API, just as Turbine switches to Repair when missing packets.